ua_parser/__init__.py

```diff
@@ -1 +1,201 @@
+import abc
+from collections.abc import Iterable
+from dataclasses import dataclass
+from typing import *
+from typing import IO
+
 VERSION = (0, 15, 0)
+
+
+@dataclass
+class UserAgent:
+    family: str
+    major: Optional[str] = None
+    minor: Optional[str] = None
+    patch: Optional[str] = None
```

Comment:
Need to look at dataclass customisations: if we can `repr()`-out the unset fields, it might be possible to use these & the yaml loading that's todo in the setup's compiler / codegen.